pkg/test/utils.go

```diff
@@ -31,3 +31,18 @@ func RemoveNodeClassTagValidation(crds []*apiextensionsv1.CustomResourceDefiniti
 	}
 	return crds
 }
+
+// DisableCapacityReservationIDValidation updates the regex validation used for capacity reservation IDs to allow any
+// string after the "cr-" prefix. This enables us to embed useful debugging information in the reservation ID, such as
+// the instance type and zone.
+func DisableCapacityReservationIDValidation(crds []*apiextensionsv1.CustomResourceDefinition) []*apiextensionsv1.CustomResourceDefinition {
+	for _, crd := range crds {
+		if crd.Name != "ec2nodeclasses.karpenter.k8s.aws" {
+			continue
+		}
+		idProps := crd.Spec.Versions[0].Schema.OpenAPIV3Schema.Properties["spec"].Properties["capacityReservationSelectorTerms"].Items.Schema.Properties["id"]
+		idProps.Pattern = `^cr-.+$`
```

Comment:
Maybe a nit, but this doesn't really "disable" the validation so the function name is sort-of misleading -- you probably honestly don't need the validation at all in those tests so you're probably fine to drop it

Reply:
Yeah, that's a good point, we can probably just drop it outright. I'm still going to prefix the test CRs though because that just feels right 🤷